cli/internal/cmd/iamcreate.go

```diff
@@ -396,6 +396,10 @@ func (c *awsIAMCreator) parseFlagsAndSetupConfig(cmd *cobra.Command, flags iamFl
 	if err != nil {
 		return iamFlags{}, fmt.Errorf("parsing zone string: %w", err)
 	}
+	// check that zone has availability zone suffix
+	if !strings.HasSuffix(zone, "a") && !strings.HasSuffix(zone, "b") && !strings.HasSuffix(zone, "c") {
+		return iamFlags{}, fmt.Errorf("missing availability zone in %s flag. Availability zone is needed to support the default attestation variant SEV-SNP. please refer to our docs and https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones", zone)
+	}
```

Comment:
Should we 

a) Make this comment more generic (not specify SEV-SNP) 

or 

b) Check this only if SEV-SNP attestation is used (which should be the default soon?)

I'm not completely sure about the state of the attestation variants on AWS, but as we will keep supporting NitroTPM attestation, this error may be misunderstood when a user creates a cluster with NitroTPM (which should work without specifying a zone iirc)

Reply:
> Should this check be at config validation instead of here?

I agree in principle to better have logic at one place. However, there is no validation check for this cmd and the current validator is imo not nice to use. I think this is sth to improve in the future in general. As we can see the validation checks are not enforced everywhere. I think it's better to fail early in the code and config validation would happen already quite late in the code path.

I would keep as is, also in sight of soonish support of multiple availability zones in the soon future according to @malt3. What do you think?

Reply:
Don't have any strong opinion here, but just as a heads-up; It would probably make sense to sync this with #1897, as the config update should probably work both with specifying the zone as a flag and having it in the config file that will be updated through `iam create --update-config` before executing the command. (i.e. users can specify the zone ad-hoc on `iam create` or specify it in their Constellation config beforehand)

Reply:
thanks, didn't know about the relevant PR

Reply:
I would like to keep validation in the config validation for now. I am happy to make up a plan on how we can improve the config validation. And once we have that plan I am also all for implementing it incrementally. But I don't want to introduce random deviations from our current way of doing this. 

This check seems like a validation that should happen on _all_ configs. Not just the ones created by `iam create`.

re "late in the code path": as far as I can see it, the Config would be created/validated before any slow operations (the IAM creation) are triggered.